core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -204,11 +203,6 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
   private val _shareCoordinatorConfig = new ShareCoordinatorConfig(this)
   def shareCoordinatorConfig: ShareCoordinatorConfig = _shareCoordinatorConfig
 
-  private val _transactionStateManagerConfig = new TransactionStateManagerConfig(this)
-  private val _addPartitionsToTxnConfig = new AddPartitionsToTxnConfig(this)
-  def transactionStateManagerConfig: TransactionStateManagerConfig = _transactionStateManagerConfig
```

Comment:
we don't hold a `TransactionLogConfig` forever now, so it seems `TransactionLogConfig` can be a POJO, right? it does not need to "hold" a `KafkaConfig"

Reply:
@chia7712, I assume what you mentioned here is `TransactionStateManagerConfig` and `AddPartitionsToTxnConfig`, right? I changed them to POJO. Thanks for your suggestion.

Reply:
@FrankYang0529 sorry for my unclear comment. My point was `TransactionLogConfig` should not hold `KafkaConfig` anymore. I will file a jira for it.

`TransactionStateManagerConfig` and `AddPartitionsToTxnConfig` constructor should take `AbstractConfig` as input for consistency. Sorry for my unclear comment again. Could you please revert the changes?

Reply:
Reverted related change. Thank you.